frontend/src/components/pages/VerifyEmail.tsx

```diff
@@ -1,21 +1,78 @@
-import React, { useContext, useEffect } from "react";
-import { Flex, Image, VStack, Center, Text } from "@chakra-ui/react";
-import { useQuery } from "@apollo/client";
+import React, {
+  Dispatch,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { Flex, Image, VStack, Center, Text, Button } from "@chakra-ui/react";
+import { useQuery, useMutation } from "@apollo/client";
 import { Redirect } from "react-router-dom";
 import RHSLogo from "../../assets/RHSlogo.png";
 import AuthContext from "../../contexts/AuthContext";
 import { GET_EMAIL_VERIFIED_BY_EMAIL } from "../../APIClients/queries/UserQueries";
 import { MANAGE_COURSES_PAGE } from "../../constants/Routes";
 import authAPIClient from "../../APIClients/AuthAPIClient";
+import { SEND_EMAIL_VERIFICATION_LINK } from "../../APIClients/mutations/AuthMutations";
+
+interface VerifyEmailProps {
+  hasError: boolean;
+  hasBeenPressed: boolean;
+  setHasBeenPressed: Dispatch<SetStateAction<boolean>>;
+}
+
+const VerifyEmailContent = ({
+  hasError,
+  hasBeenPressed,
+  setHasBeenPressed,
+}: VerifyEmailProps): React.ReactElement<VerifyEmailProps> => {
+  if (hasError) {
+    return (
+      <Text variant="display-md" paddingBottom="1vw">
+        An error has occurred while resending your verification email
+      </Text>
+    );
+  }
+  if (hasBeenPressed) {
+    return (
+      <Text variant="display-md" paddingBottom="1vw">
+        Verification email has been sent
+      </Text>
+    );
+  }
+  return (
+    <>
+      <Text variant="display-md" paddingBottom="1vw">
+        Please verify your email to continue
+      </Text>
+      <Button
+        variant="sm"
+        width="full"
+        marginBottom="2vh"
+        onClick={() => setHasBeenPressed(true)}
+      >
+        Resend verification email
+      </Button>
+    </>
+  );
+};
 
 const VerifyEmail = (): React.ReactElement => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+
+  const [hasBeenPressed, setHasBeenPressed] = useState(false);
+  const [hasError, setHasError] = useState(false);
```

Comment:
Rather than having several different flags, it could make sense to just keep one "state" variable since these states are all mutually exclusive. It might make the logic clearer (i.e., we're either in the "default" state, the "sent" state or the "error" state). Just an optional suggestion though, I don't mind this solution as is since there are only two flags.

Reply:
Yeah, I thought this too, definitely better logically to change it, but I didn't think about it much more.